temp_functions.py

```diff
@@ -0,0 +1,75 @@
+import read_db
+import pandas as pd
+import time
+from tqdm import tqdm
+
+class functions():
+
+    def __init__(self):
+        self.df=None
+        self.dict={}
+
+    def set_dataframe(self):
+        extract = read_db.ExtractData()
+        self.df = extract.get_timeslots_info()
+
+    def main(self):
+        self.set_dataframe()
+        temp = {}
+        for i,row in tqdm(self.df.iterrows()):
+            key = row["EmployeeID"], row["RelationID"]
+
+            ## Calculate the remaining availability in each calendar month and week
+            appointmentLength = (row["UntilUtc"] - row["FromUtc"]).total_seconds()/3600
+            if key[0] in temp:
+                if temp[key[0]]["DateLastWorked"].year == row["FromUtc"].year:
+                    if temp[key[0]]["DateLastWorked"].month == row["FromUtc"].month:
+                        temp[key[0]]["HoursLeftInMonth"] = temp[key[0]]["HoursLeftInMonth"] - appointmentLength
+                    else: 
+                        temp[key[0]]["HoursLeftInMonth"] = row["AverageNumberOfHoursPerMonth"] - appointmentLength
+                    if temp[key[0]]["DateLastWorked"].isocalendar().week == row["FromUtc"].isocalendar().week:
+                        temp[key[0]]["HoursLeftInWeek"] = temp[key[0]]["HoursLeftInWeek"] - appointmentLength
```

Comment:
Wat is het nut om wekelijkse uren te pakken? Contract geef maandelijkse uren toch?

Reply:
> Wat is het nut om wekelijkse uren te pakken? Contract geef maandelijkse uren toch?

Hier twijfelde ik ook over, maar het was niet moeilijk om toe te voegen dus ik heb het maar snel gedaan. Mogelijk dat dit iets zou kunnen helpen om de employees iets beter over de weken te verdelen.

Reply:
Ja hou het er maar even in, kunnen we altijd later eruithalen